examples/aggregator/module/rust/src/tests.rs

```diff
@@ -46,31 +58,103 @@ fn test_aggregator() {
     let (runtime, entry_channel) = oak_tests::run_single_module_default(MODULE_CONFIG_NAME)
         .expect("Unable to configure runtime with test wasm!");
 
-    submit_sample(&runtime, &entry_channel, vec![0, 1, 0, 1, 0]);
-    submit_sample(&runtime, &entry_channel, vec![1, 0, 1, 0, 1]);
-    {
-        let req = Empty::new();
-        let result: grpc::Result<Vector> = oak_tests::grpc_request(
-            &runtime,
-            &entry_channel,
-            "/oak.examples.aggregator.Aggregator/GetCurrentValue",
-            &req,
+    for i in 0..SAMPLE_THRESHOLD as u32 {
+        assert_matches!(
+            submit_sample(
+                &runtime,
+                &entry_channel,
+                "test",
+                vec![i, i + 1, i + 2],
+                vec![10.0, 20.0, 30.0]
+            ),
+            Ok(_)
         );
-        assert_matches!(result, Err(_));
     }
-
-    submit_sample(&runtime, &entry_channel, vec![1, 1, 1, 1, 1]);
-    {
-        let req = Empty::new();
-        let result: grpc::Result<Vector> = oak_tests::grpc_request(
+    // After sending the `SAMPLE_THRESHOLD` of samples, it's not possible to use the same `bucket`.
+    assert_matches!(
+        submit_sample(
             &runtime,
             &entry_channel,
-            "/oak.examples.aggregator.Aggregator/GetCurrentValue",
-            &req,
-        );
-        assert_matches!(result, Ok(_));
-        assert_eq!(vec![2, 2, 2, 2, 2], result.unwrap().items);
-    }
+            "test",
+            vec![1, 2, 3],
+            vec![10.0, 20.0, 30.0]
+        ),
+        Err(_)
+    );
 
     runtime.stop();
 }
+
+fn check_combine(src1: HashMap<u32, f32>, src2: HashMap<u32, f32>, dst: HashMap<u32, f32>) -> bool {
+    let src = SparseVector::new(src1).combine(&SparseVector::new(src2));
+    src == SparseVector::new(dst)
```

Comment:
May as well do the assert here :) otherwise, if you had more things to check, it would not be feasible to return everything, and expect the caller would just assert all of them.

An alternative would be to write a table-driven test, e.g.

```rust
#[test]
fn test_combine() {
  type Map = HashMap<u32, f32>;
  struct Test {
    in_0: Map,
    in_1: Map,
    out: Map,
  }
  let tests = vec![
    Test {
      in_0: hashmap! {},
      in_1: hashmap! { 1 => 10.0 },
      out: hashmap! { 1 => 10.0 },
    },
    ...
  ];
  for test in tests {
    ...
  }
}
```

and also below you could use the same approach for other tests.

WDYT?